### PR TITLE
Fixes table content overflowing

### LINKED ISSUE #10517: after v3.1.23 Table content is overflowing
### Package

filament/filament

### Package Version

v3.1.29

### Laravel Version

v10.38.1

### Livewire Version

_No response_

### PHP Version

PHP 8.2.12

### Problem description

Tables overflow in screen
Screenshot 2023-12-27 at 11 21 59 am

Due to this PR on v3.1.23 [https://github.com/filamentphp/filament/pull/10290](https://github.com/filamentphp/filament/pull/10290)

### Expected behavior

table should not overflow
Screenshot 2023-12-27 at 1 00 04 pm)

### Steps to reproduce

upgrade any version above v3.1.23

### Reproduction repository

https://github.com/danieladarve/filament

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by danieladarve**
Closes  [#10517](https://github.com/filamentphp/filament/issues/10517)

### CODE REVIEW

**Comment by bmckay959** [comment]
@danieladarve - By adding `relative` back, it breaks a menu that has tenancy and collapsed.

I'm not able to replicate on my side for some reason. Do you have tenancy and collapsible menu on your project?

You can see the issue here: https://github.com/filamentphp/filament/pull/10290

**Comment by danieladarve** [comment]
@bmckay959  I added it to the actual table not the containing div where it was removed from.

No i do not have tenancy. But the same occurs I enable the collapsible menu.
That menu that you mentioned could be fixed adding a z-index to the table element and the menu element.

**Comment by danieladarve** [comment]
that menu could be fixed adding a z-index to the element

**Comment by danieladarve** [comment]
@bmckay959 the issue doesn't seem to happen if you dont have any sortable columns. Did you test with sortable columns?

**Comment by bmckay959** [comment]
Nope. My project doesn't have sortable columns. Let me add for testing and I'll get back here

**Comment by bmckay959** [comment]
@danieladarve - Just to confirm, when you remove `->sortable()` from the table, the issue goes away?

**Comment by danieladarve** [comment]
if you dont have any sortable columns then the table doesnt overflow.

**Comment by danieladarve** [comment]
@bmckay959 See [commit](https://github.com/filamentphp/filament/pull/10518/commits/e08c9f16ec5ffc0ee2f55894bbd4565f38638e8a)

It fixes your issue  [#10290](https://github.com/filamentphp/filament/pull/10290) by using z-index in the sidebar
And also fixes the table content overflowing issue!!

Here are some screenshots testing different states of the side menu

Screenshot 2023-12-27 at 5 15 14 pm

Screenshot 2023-12-27 at 5 21 04 pm

Screenshot 2023-12-27 at 5 21 24 pm

Screenshot 2023-12-27 at 5 24 06 pm

Screenshot 2023-12-27 at 5 25 11 pm

**Comment by KarelBrijs** [comment]
@bmckay959 @danieladarve this does seem to resolve it for me!

**Comment by bmckay959** [comment]
Nice!

I'm going to do a bit more digging this morning. My gut is telling me that if it only happens with sorting, then maybe something weird is going on in that portion, but if not, it looks like this resolved it for everyone.

Do either of you have table header widgets to confirm those are also behind the dropdown? I'm trying to think of what else this could be opening up...

**Comment by bmckay959** [comment]
I just pulled in table widgets and can confirm, this appears to solve all of the issues.

Thanks @KarelBrijs and @danieladarve !

@zepfietje - I think we are good for you to review.

File: packages/panels/resources/views/components/sidebar/index.blade.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
         @endif
     @endif
     @class([
-        'fi-sidebar fixed inset-y-0 start-0 z-30 grid h-screen content-start bg-white transition-all dark:bg-gray-900 lg:z-0 lg:bg-transparent lg:shadow-none lg:ring-0 lg:transition-none dark:lg:bg-transparent',
+        'fi-sidebar fixed inset-y-0 start-0 z-30 grid h-screen content-start bg-white transition-all dark:bg-gray-900 lg:bg-transparent lg:shadow-none lg:ring-0 lg:transition-none dark:lg:bg-transparent',
```
Comment (zepfietje) [inline packages/panels/resources/views/components/sidebar/index.blade.php:37]: Why aren't we changing `z-30` here to `z-40` instead?
Haven't tested, but I'm just asking to prevent breaking any edge cases.

Also, have you tested this change with modals?
Modals also have `z-40`, so make sure this PR doesn't break anything there. 🙂

Since I don't have a simple reproduction repository, I can't actually test the change.

Comment (danieladarve) [reply packages/panels/resources/views/components/sidebar/index.blade.php:37]: @zepfietje you are right it didnt need to change. It can stay as ``z-30``. What made the difference was removing ``lg:z-0``. Just made a test with both ``z-30`` and ``z-40`` and the modals look okay.

Here are some screenshots

Screenshot 2023-12-28 at 8 15 10 am
Screenshot 2023-12-28 at 7 48 26 am
Screenshot 2023-12-28 at 8 18 10 am

I reverted back to ``z-30`` to have less changes in this PR
Will upload the reproduction repo in just a moment

Comment (danieladarve) [reply packages/panels/resources/views/components/sidebar/index.blade.php:37]: @zepfietje Here is the repo

username: admin@admin.com
password: password

https://github.com/danieladarve/filament-overflow-issue

File: packages/panels/resources/views/components/sidebar/index.blade.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
         x-bind:class="
             $store.sidebar.isOpen
                 ? @js($openSidebarClasses . ' ' . 'lg:sticky')
-                : '-translate-x-full rtl:translate-x-full lg:sticky lg:translate-x-0 rtl:lg:-translate-x-0'
+                : 'z-30 -translate-x-full rtl:translate-x-full lg:sticky lg:translate-x-0 rtl:lg:-translate-x-0'
```
Comment (zepfietje) [inline packages/panels/resources/views/components/sidebar/index.blade.php:18]: `z-30` isn't needed any longer here, right?

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 1 above)
Comment (zepfietje) [inline packages/panels/resources/views/components/sidebar/index.blade.php:37]: I feel like there must be a reason why `lg:z-0` was used initially... Have you tested all sidebar configurations (collapsible, fully collapsible, etc.)?

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 1 above)
Comment (zepfietje) [reply packages/panels/resources/views/components/sidebar/index.blade.php:37]: Thanks, @danieladarve!

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 2 above)
Comment (danieladarve) [reply packages/panels/resources/views/components/sidebar/index.blade.php:18]: It is otherwise the tenancy menu will go under the table.

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 1 above)
Comment (danieladarve) [reply packages/panels/resources/views/components/sidebar/index.blade.php:37]: @zepfietje yes I did, I also provided screenshots for the different sidebar states/modes, including when a modal is active.

I am unsure what the purpose of removing z-index in larges screens is for. I looked up the commit history and it goes all the way back to 2.x branch.

**Comment by bmckay959** [comment]
Does it feel like maybe we are looking in the wrong place?

If the only time we have an issue is when a table has sortable columns, then maybe it's a wrong class in the sortable column type? I'm going to look there again

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 2 above)
Comment (zepfietje) [reply packages/panels/resources/views/components/sidebar/index.blade.php:18]: But `z-30` is already in `@class()` below here?

File: packages/panels/resources/views/components/sidebar/index.blade.php (hunk 1 above)
Comment (zepfietje) [reply packages/panels/resources/views/components/sidebar/index.blade.php:37]: Hmm alright, let's hope it doesn't break anything then...

**Comment by tareq-alqadi** [comment]
I faced same problem in table with sortable columns.
The problem was in span which have "sr-only" class specifically in

```
width: 1px;
height: 1px;
```
I changed them to
```
width: 0px;
height: 0px;
```
and the problem was fixed.

**Comment by danharrin** [comment]
Can someone please create a reproduction repository? The one in the original issue is not a reproduction repository, just a fork of Filament. And I have downloaded https://github.com/danieladarve/filament-overflow-issue, and the tables do not scroll for me. Please create an app from scratch with a seeder that contains the data that scrolls the table. Please do not include other features that are not relevant to the issue.

**Comment by danharrin** [comment]
And if the fix is to instead replace `sr-only` elements with `aria-label`, please do so in a separate PR and close this one.

**Comment by danharrin** [comment]
Looks like there is a new issue, hopefully the reproduction is good: #10633

**Comment by danharrin** [comment]
Yup, it was good. Fixed in #10636.
